src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -145,7 +145,10 @@ public void persist() throws IOException {
         final Path path = Paths.get(fileName);
         final Path directory = path.getParent();
         if (directory != null) {
-            Files.createDirectories(directory);
+            // createDirectories throws an error if the target directory exists and is a symbolic link,
+            // skip call if the directory already exists
+            if (!directory.toFile().isDirectory())
+                Files.createDirectories(directory);
```

Comment:
Please add test on this, it is ok to have it only Linux specific

Reply:
Java has special method to check if file is a link https://docs.oracle.com/javase/tutorial/essential/io/links.html#:~:text=To%20determine%20whether%20a%20Path,isSymbolicLink(file)%3B

Better to use it.